src/GraphQL.Tests/Bugs/Bug1773.cs

```diff
@@ -0,0 +1,116 @@
+        }
+
+        [Fact]
+        public void list_throws_for_invalid_type()
+        {
+            // TODO: does not yet fully meet spec (does not return members of lists that are able to be serialized, with nulls and individual errors for unserializable values)
+            AssertQueryWithError("{testListInvalidType}", "{\"testListInvalidType\": null}", "Error trying to resolve testListInvalidType.", 1, 2, new[] { "testListInvalidType" },
+                new FormatException("Input string was not in a correct format."));
+        }
+
+        [Fact]
+        public void list_throws_for_invalid_type_when_conversion_returns_null()
+        {
+            // TODO: does not yet fully meet spec (does not return members of lists that are able to be serialized, with nulls and individual errors for unserializable values)
+            AssertQueryWithError("{testListInvalidType2}", "{\"testListInvalidType2\": null}", "Error trying to resolve testListInvalidType2.", 1, 2, new[] { "testListInvalidType2" },
+                new InvalidOperationException("Unable to serialize 'test' to 'Bug1773Enum' for list index 0."));
+        }
+
+        [Fact]
+        public void throws_for_invalid_type()
+        {
+            // in this case, the conversion threw a FormatException
+            AssertQueryWithError("{testInvalidType}", "{\"testInvalidType\": null}", "Error trying to resolve testInvalidType.", 1, 2, new[] { "testInvalidType" },
+                new FormatException("Input string was not in a correct format."));
+        }
+
+        [Fact]
+        public void throws_for_invalid_type_when_conversion_returns_null()
+        {
+            // in this case, the converstion returned null, and GraphQL threw an InvalidOperationException
+            AssertQueryWithError("{testInvalidType2}", "{\"testInvalidType2\": null}", "Error trying to resolve testInvalidType2.", 1, 2, new[] { "testInvalidType2" },
+                new InvalidOperationException("Unable to serialize 'test' to 'Bug1773Enum'"));
+        }
+
+    }
+
+    public class Bug1773Schema : Schema
+    {
+        public Bug1773Schema()
+        {
+            Query = new Bug1773Query();
+        }
+    }
+
+    public class Bug1773Query : ObjectGraphType
+    {
+        public Bug1773Query()
+        {
+            Field<ListGraphType<IntGraphType>>("testListValid", resolve: context => new object[] { 123 });
+            Field<ListGraphType<IntGraphType>>("testListInvalid", resolve: context => 123);
+            Field<ListGraphType<IntGraphType>>("testListInvalidType", resolve: context => new object[] { "test" });
+            Field<ListGraphType<EnumerationGraphType<Bug1773Enum>>>("testListInvalidType2", resolve: context => new object[] { "test" });
+            Field<ListGraphType<NonNullGraphType<IntGraphType>>>("testListNullValid", resolve: context => new object[] { 123 });
```

Comment:
```suggestion
            Field<ListGraphType<NonNullGraphType<IntGraphType>>>("testListNonNullValid", resolve: context => new object[] { 123 });
```

and so on for other non null fields, otherwise the meaning is inverted

Reply:
Nevermind, I'm OK with current naming.